### ISSUE TITLE
ButterflyCount - capture precise location for every individual butterfly seen

### ISSUE THREAD

**Issue by DavidRoy** [BiologicalRecordsCentre/ABLE#237]
Currently, there butterfly count is assigned to a survey area (GPS track or area drawn on a map).  There is a require to capture precise location information for every individual butterfly seen (e.g. when a species is added to the sample list and when the numbers are increased).

The expectation from users is to have precise location information

**Comment by kazlauskis** [comment]
We have had a brief discussion of this last year. One of the issues we have come across was knowing which species record to drop from the list when decrementing a count. 

To properly accommodate subsample recording this would require big changes to the interface.

**Comment by DavidRoy** [comment]
It would be the last occurrence to be deleted if decrementing a count

**Comment by kazlauskis** [comment]
I can do it but what about changing a species ID. Say you have realised that one of the first of the 5 Large Whites you have recorded was actually a Small White, when decrementing would it still be OK to drop the last one?

**Comment by DavidRoy** [comment]
After discussing options, the next steps are:
- karolis to mockup a proposed User Interface
- david to consider storage implications

**Comment by chrisvanswaay** [comment]
I would keep the normal input screen as it is (it works really fine in the field). If recorders want to make a change, they should go to a special screen in which they see all their entries one by one. That way you can delete records, or make a change: select one record, tap it, and make the change.
Indeed something like this (or something else) would require an extra screen.
It will also mean I will have to stop promoting the app in NL, so far I said it would become available soon, and many were 'on hold' waiting for that, but now it will mean the app cannot be used for professional inventories (as e.g. on Natura 2000 areas in NL, normally done in great precision). 
It is how it is, I realise there is no unlimited funding.
Thanks for considering.

**Comment by kazlauskis** [comment]
I will do the mockup and will add it here shortly.

**Comment by kazlauskis** [comment]
@chrisvanswaay @DavidRoy Sorry for the delay. 

Overall, this is quite a big structural change to how the data is organised. One way would be to wrap each occurrence within a sample and assign it precise locations from the GPS. The number +/- buttons would only append/delete the last species entry in the list, but you could manually swipe left any occurrence and delete it. You could also change the ID of an occurrence by opening it up and changing the species. It would then occur in another list for that species. One thing to note here is that we have moved the stage attribute to each of the occurrences edit pages. 

Here is the interactive mockup link, please share your thoughts: https://projects.invisionapp.com/prototype/eBMS-App-ckc3bwfcw006hob01vqrk4fmr/play/0f118c94

![Screenshot 2020-07-01 at 13 43 42](https://user-images.githubusercontent.com/7383162/86245118-ee931700-bba0-11ea-8f7b-bbe6264d3e53.png)
![Screenshot 2020-07-01 at 13 43 47](https://user-images.githubusercontent.com/7383162/86245116-ed61ea00-bba0-11ea-832a-c8d38fa2ee9f.png)


![Screenshot 2020-07-01 at 13 43 51](https://user-images.githubusercontent.com/7383162/86245113-eb982680-bba0-11ea-9a43-79b8fe2f0ca5.png)

**Comment by chrisvanswaay** [comment]
Hi Karolis,

Thanks for this. Looks good, though I can't seem to test it. I have had
this problem before, not sure why. But the solution to show the location
and time of the observation seems logical. The only thing that might help
is to delete from the map with time (so the map of the track with the
observations as points with the species name next to it?). But I guess that
could potentially be a lot of work, so this solution will also do.

Best wishes,
Chris

--

*De Vlinderstichting is een organisatie zonder winstoogmerk. Voor een deel
van ons werk zijn wij afhankelijk van de steun van donateurs. Wilt u
bijdragen aan bescherming van vlinders en libellen? Neem dan een
kijkje op* *onze
website* <https://www.vlinderstichting.nl/help-mee/donateur-worden> *en
help ons!*

Chris van Swaay
De Vlinderstichting
Mennonietenweg 10
Postbus 506
6700 AM  Wageningen

[t] algemeen (+31) 0317 467346 / direct (+31) 0317 467342

[e] chris.vanswaay@vlinderstichting.nl
[i] www.vlinderstichting.nl
[twitter] www.twitter.com/vlinderNL
[facebook] www.facebook.com/Vlinderstichting
[instagram] www.instagram.com/vlinderstichting
[linkedin] www.linkedin.com/company/vlinderstichting

------------------------------
Aan de inhoud van dit e-mailbericht kunnen geen rechten worden ontleend. De
informatie is uitsluitend bestemd voor de geadresseerde. De
Vlinderstichting staat niet in voor de juiste en volledige overbrenging van
de inhoud van een verzonden e-mailbericht noch voor tijdige aankomst
daarvan.
-----------------------------
No rights may be claimed from the contents of this e-mail. The information
is only intended for the addressee. Dutch Butterfly Conservation cannot be
held responsible for the correctness and completeness of the contents of an
e-mail, nor for its arrival on time.


Op wo 1 jul. 2020 om 14:47 schreef Karolis Kazlauskis <
notifications@github.com>:

**Comment by kazlauskis** [comment]
Yes, I had a thought like this. It would be great to show all the occurrences on the map and would help with the validation of locations. I suggest to wait until this basic functionality is tested a bit more in the field before adding the extra mapping. 

A question I have is whether you would want to correct/set any occurrence locations manually? At the moment, it is GPS-only but we could add a map or a lat/long input box to change it.

**Comment by chrisvanswaay** [comment]
Thanks Karolis. Good question, also as so far I have had to redraw the line
often, as my gps is not very good and when you don't see any butterflies
and have it in your pocket, it loses track of where you are. So this will
also depend on such things which we can't all influence. But I guess people
would want to do that on a map, I guess no one understands coordinates. I
think I would like to move an observation on the map.

Best wishes
Chris

--

*De Vlinderstichting is een organisatie zonder winstoogmerk. Voor een deel
van ons werk zijn wij afhankelijk van de steun van donateurs. Wilt u
bijdragen aan bescherming van vlinders en libellen? Neem dan een
kijkje op* *onze
website* <https://www.vlinderstichting.nl/help-mee/donateur-worden> *en
help ons!*

Chris van Swaay
De Vlinderstichting
Mennonietenweg 10
Postbus 506
6700 AM  Wageningen

[t] algemeen (+31) 0317 467346 / direct (+31) 0317 467342

[e] chris.vanswaay@vlinderstichting.nl
[i] www.vlinderstichting.nl
[twitter] www.twitter.com/vlinderNL
[facebook] www.facebook.com/Vlinderstichting
[instagram] www.instagram.com/vlinderstichting
[linkedin] www.linkedin.com/company/vlinderstichting

------------------------------
Aan de inhoud van dit e-mailbericht kunnen geen rechten worden ontleend. De
informatie is uitsluitend bestemd voor de geadresseerde. De
Vlinderstichting staat niet in voor de juiste en volledige overbrenging van
de inhoud van een verzonden e-mailbericht noch voor tijdige aankomst
daarvan.
-----------------------------
No rights may be claimed from the contents of this e-mail. The information
is only intended for the addressee. Dutch Butterfly Conservation cannot be
held responsible for the correctness and completeness of the contents of an
e-mail, nor for its arrival on time.


Op wo 1 jul. 2020 om 17:01 schreef Karolis Kazlauskis <
notifications@github.com>:

**Comment by kazlauskis** [comment]
@chrisvanswaay we have built the functionality to capture precise locations for individual species during the 15min count. This includes the option to correct any occurrence locations using a map as well. The functionality isn't yet available to the public and it would be great if you had a look at how it works and let us know if you're happy with it. Here is a video to illustrate the functionality: https://youtu.be/NRxG4gNeSPM

**Comment by chrisvanswaay** [comment]
Hi Karolis,

Great that you managed to work on this. The youtube video looks good, this looks like a perfect solution. I can't test it on my own phone.

Best wishes,
Chris


--

De Vlinderstichting is een organisatie zonder winstoogmerk. Voor een deel van ons werk zijn wij afhankelijk van de steun van donateurs. Wilt u bijdragen aan bescherming van vlinders en libellen? Neem dan een kijkje op onze website<https://www.vlinderstichting.nl/help-mee/donateur-worden> en help ons!


Chris van Swaay
De Vlinderstichting
Mennonietenweg 10
Postbus 506
6700 AM  Wageningen

[t] algemeen (+31) 0317 467346 / direct (+31) 0317 467342

[e] chris.vanswaay@vlinderstichting.nl<mailto:chris.vanswaay@vlinderstichting.nl>
[i] www.vlinderstichting.nl<http://www.vlinderstichting.nl/>
[twitter] www.twitter.com/vlinderNL<http://www.twitter.com/vlinderNL>
[facebook] www.facebook.com/Vlinderstichting<http://www.facebook.com/Vlinderstichting>
[instagram] www.instagram.com/vlinderstichting<http://www.instagram.com/vlinderstichting>
[linkedin] www.linkedin.com/company/vlinderstichting<http://www.linkedin.com/company/vlinderstichting>

------------------------------
Aan de inhoud van dit e-mailbericht kunnen geen rechten worden ontleend. De informatie is uitsluitend bestemd voor de geadresseerde. De Vlinderstichting staat niet in voor de juiste en volledige overbrenging van de inhoud van een verzonden e-mailbericht noch voor tijdige aankomst daarvan.
-----------------------------
No rights may be claimed from the contents of this e-mail. The information is only intended for the addressee. Dutch Butterfly Conservation cannot be held responsible for the correctness and completeness of the contents of an e-mail, nor for its arrival on time.

________________________________
Van: Karolis Kazlauskis <notifications@github.com>
Verzonden: zaterdag 24 oktober 2020 11:51
Aan: BiologicalRecordsCentre/ABLE <ABLE@noreply.github.com>
CC: Chris van Swaay <Chris.vanSwaay@vlinderstichting.nl>; Mention <mention@noreply.github.com>
Onderwerp: Re: [BiologicalRecordsCentre/ABLE] ButterflyCount - capture precise location for every individual butterfly seen (#237)


@chrisvanswaay<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fchrisvanswaay&data=04%7C01%7Cchris.vanswaay%40vlinderstichting.nl%7C98251c38b4834aaf3fac08d878026aed%7C94e10a529c02479bb9e5efb937389e5f%7C1%7C0%7C637391299066174620%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=%2FferU6cbvpCZDgdKhzVmPtYRs03k6Sxjtr%2BLD16ZRqk%3D&reserved=0> we have built the functionality to capture precise locations for individual species during the 15min count. This includes the option to correct any occurrence locations using a map as well. The functionality isn't yet available to the public and it would be great if you had a look at how it works and let us know if you're happy with it. Here is a video to illustrate the functionality: https://youtu.be/NRxG4gNeSPM<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fyoutu.be%2FNRxG4gNeSPM&data=04%7C01%7Cchris.vanswaay%40vlinderstichting.nl%7C98251c38b4834aaf3fac08d878026aed%7C94e10a529c02479bb9e5efb937389e5f%7C1%7C0%7C637391299066174620%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=HA1mb8Z6c02oH1XeNoP%2FHX%2FcSE70tWUpY%2BcscqMQPPk%3D&reserved=0>

**Comment by kazlauskis** [comment]
From Chris:
> Thanks Karolis, it worked as it should. Though still doesn't feel very intuitive (sorry). What I somehow expected was that when I click the species, I would get the observations of that species on the map. Because most people won't recognize the latlong or the time of their record (which is what you now get), but do remember the place. Then clicking one of the dots offer the option to change. What do you think?

**Comment by kazlauskis** [comment]
You can click on one of the dots in the overall survey area map, this takes you directly to the occurrence where you can edit it then. We could add a map to the species occurrences page as well. The issue with maps is that it requires network connection to be any useful. If we removed the list and replaced it with a map instead then we would assume our users have the connection so we should probably have both options.

**Comment by chrisvanswaay** [comment]
Hi Karolis,

One important thing is still missing in the app (though I reported that earlier), at least, as far as I can see.
It is now possible to enter records from multiple species groups. However we have to be sure that ALL species of a group are entered. Lets assume I do a transect counting butterflies, and I also recognize (among many other dragonflyspecies flying around as well) the only species of dragonfly I can identify, and enter that as well. But not all the other dragonflies (which I can't identify). Then this might end up as a dragonfly count with only 1 species. So we have to find a way to be sure that all species are recognized and entered. Two possible ways this could be done:

  *   Before starting: you have to choose the species groups you want to count. If you choose butterflies and dragonflies, you won't be able to enter another group. And if you choose only butterlfies, you only get butterflies when you search for a species, so you can't enter a dragonfly anymore (or alternatively: they will be marked as single opportunistic records)
  *   At the end: for which species groups have you entered ALL you have seen?

I would prefer the first, and make other species opportunistic records.

Best wishes,
Chris


--

De Vlinderstichting is een organisatie zonder winstoogmerk. Voor een deel van ons werk zijn wij afhankelijk van de steun van donateurs. Wilt u bijdragen aan bescherming van vlinders en libellen? Neem dan een kijkje op onze website<https://www.vlinderstichting.nl/help-mee/donateur-worden> en help ons!


Chris van Swaay
De Vlinderstichting
Mennonietenweg 10
Postbus 506
6700 AM  Wageningen

[t] algemeen (+31) 0317 467346 / direct (+31) 0317 467342

[e] chris.vanswaay@vlinderstichting.nl<mailto:chris.vanswaay@vlinderstichting.nl>
[i] www.vlinderstichting.nl<http://www.vlinderstichting.nl/>
[twitter] www.twitter.com/vlinderNL<http://www.twitter.com/vlinderNL>
[facebook] www.facebook.com/Vlinderstichting<http://www.facebook.com/Vlinderstichting>
[instagram] www.instagram.com/vlinderstichting<http://www.instagram.com/vlinderstichting>
[linkedin] www.linkedin.com/company/vlinderstichting<http://www.linkedin.com/company/vlinderstichting>

------------------------------
Aan de inhoud van dit e-mailbericht kunnen geen rechten worden ontleend. De informatie is uitsluitend bestemd voor de geadresseerde. De Vlinderstichting staat niet in voor de juiste en volledige overbrenging van de inhoud van een verzonden e-mailbericht noch voor tijdige aankomst daarvan.
-----------------------------
No rights may be claimed from the contents of this e-mail. The information is only intended for the addressee. Dutch Butterfly Conservation cannot be held responsible for the correctness and completeness of the contents of an e-mail, nor for its arrival on time.

________________________________
Van: Karolis Kazlauskis <notifications@github.com>
Verzonden: donderdag 5 november 2020 15:13
Aan: BiologicalRecordsCentre/ABLE <ABLE@noreply.github.com>
CC: Chris van Swaay <Chris.vanSwaay@vlinderstichting.nl>; Mention <mention@noreply.github.com>
Onderwerp: Re: [BiologicalRecordsCentre/ABLE] ButterflyCount - capture precise location for every individual butterfly seen (#237)


You can click on one of the dots in the overall survey area map, this takes you directly to the occurrence where you can edit it then. We could add a map to the species occurrences page as well. The issue with maps is that it requires network connection to be any useful. If we removed the list and replaced it with a map instead then we would assume our users have the connection so we should probably have both options.

**Comment by kazlauskis** [comment]
Thanks, I have created a new ticket for choosing the species group #313
